services/block.go

```diff
@@ -103,57 +110,136 @@ func (s *blockAPIService) Block(
 		return nil, ErrUnableToGetBlk
 	}
 
-	evts, err := s.oasisClient.GetStakingEvents(ctx, height)
+	// We group transactions by hash and number the operations within a
+	// transaction in the order as they appear.
+	txns := []*types.Transaction{}
+	txnmap := make(map[string]*types.Transaction) // hash -> transaction in txns
+	// getTxn get existing transaction if it already exists or
+	// create a new one and return it if it doesn't.
+	getTxn := func(txhash string) *types.Transaction {
+		if txn, exists := txnmap[txhash]; exists {
+			return txn
+		}
+		tx := &types.Transaction{
+			TransactionIdentifier: &types.TransactionIdentifier{
+				Hash: txhash,
+			},
+			Operations: []*types.Operation{},
+		}
+		txns = append(txns, tx)
+		txnmap[txhash] = tx
+		return tx
+	}
+
+	rawTxs, err := s.oasisClient.GetTransactions(ctx, height)
 	if err != nil {
 		loggerBlk.Error("Block: unable to get transactions",
 			"height", height,
 			"err", err,
 		)
 		return nil, ErrUnableToGetTxns
 	}
-
-	// We group transactions by hash and number the operations within a
-	// transaction in the order as they appear.
-	txns := []*types.Transaction{}
-	txnmap := make(map[string]uint) // hash -> index into txns
-	for _, evt := range evts {
-		// Get index of existing transaction if it already exists or
-		// create a new one and return its index if it doesn't.
-		var txidx uint
-		txhash := evt.TxHash.String()
-		if idx, exists := txnmap[txhash]; exists {
-			txidx = idx
-		} else {
-			txidx = uint(len(txns))
-			txnmap[txhash] = txidx
-			txns = append(txns, &types.Transaction{
-				TransactionIdentifier: &types.TransactionIdentifier{
-					Hash: txhash,
+	for i, rawTx := range rawTxs {
+		var sigTx transaction.SignedTransaction
+		if err := cbor.Unmarshal(rawTx, &sigTx); err != nil {
+			loggerBlk.Warn("Block: malformed transaction",
+				"height", height,
+				"index", i,
+				"raw_tx", rawTx,
+				"err", err,
+			)
+			continue
+		}
+		var tx transaction.Transaction
+		if err := sigTx.Open(&tx); err != nil {
+			loggerBlk.Warn("Block: invalid transaction signature",
+				"height", height,
+				"index", i,
+				"sig_tx", sigTx,
+				"err", err,
+			)
+			continue
+		}
+		switch tx.Method {
+		case staking.MethodReclaimEscrow:
+			var reclaim staking.ReclaimEscrow
+			if err := cbor.Unmarshal(tx.Body, &reclaim); err != nil {
+				loggerBlk.Warn("Block: malformed reclaim escrow",
+					"height", height,
+					"index", i,
+					"body", tx.Body,
+					"err", err,
+				)
+				continue
+			}
+			// Emit the reclaim escrow intent.
+			// TODO: Maybe alter oasis-core to emit a staking event so we
```

Comment:
No need to alter, you also have `GetTransactionsWithResults` which will also get you the transaction result (error and all emitted events). Just bump to 20.9 which we should do anyway (see #30).

Reply:
oh let's do that first

Reply:
changed it to use that, thanks